pkg/k8shandler/cluster.go

```diff
@@ -141,7 +141,12 @@ func CreateOrUpdateElasticsearchCluster(cluster *v1alpha1.Elasticsearch) error {
 						return err
 					}
 
-					node.create()
+					// if we go to create the object and it has a resourceVersion already --
+					// that means it was previously created.
```

Comment:
Do you need to do a RetryOnConflict here?

Reply:
this is handled in the `node.create()` code... leftover comment. But we don't want to do a retry when we are just trying to create. For these objects we separate them explicitly because we may be required to perform cluster-wide actions first.